src/cascadia/TerminalSettingsEditor/AddProfile.xaml

```diff
@@ -0,0 +1,83 @@
+      xmlns:model="using:Microsoft.Terminal.Settings.Model"
+      xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
+      mc:Ignorable="d">
+
+    <Page.Resources>
+        <ResourceDictionary>
+            <ResourceDictionary.MergedDictionaries>
+                <ResourceDictionary Source="CommonResources.xaml" />
+            </ResourceDictionary.MergedDictionaries>
+            <model:IconPathConverter x:Key="IconSourceConverter" />
+        </ResourceDictionary>
+    </Page.Resources>
+
+    <ScrollViewer>
+        <StackPanel Style="{StaticResource SettingsStackStyle}">
+            <Button x:Name="AddNewButton"
+                    Click="AddNewClick"
+                    Style="{StaticResource AccentButtonStyle}">
+                <Button.Content>
+                    <StackPanel Orientation="Horizontal">
+                        <FontIcon FontSize="{StaticResource StandardIconSize}"
+                                  Glyph="&#xE710;" />
+                        <TextBlock x:Uid="AddProfile_AddNewButton"
+                                   Style="{StaticResource IconButtonTextBlockStyle}" />
+                    </StackPanel>
+                </Button.Content>
+            </Button>
+            <StackPanel Margin="{StaticResource StandardControlMargin}">
+                <local:SettingContainer x:Uid="AddProfile_Duplicate">
+                    <muxc:RadioButtons x:Name="Profiles"
+                                       ItemsSource="{x:Bind State.Settings.AllProfiles, Mode=OneWay}">
+                        <muxc:RadioButtons.ItemTemplate>
+                            <DataTemplate x:DataType="model:Profile">
+                                <Grid HorizontalAlignment="Stretch"
+                                      ColumnSpacing="8">
+
+                                    <Grid.ColumnDefinitions>
+                                        <!--  icon  -->
+                                        <ColumnDefinition Width="16" />
+                                        <!--  profile name  -->
+                                        <ColumnDefinition Width="Auto" />
+                                    </Grid.ColumnDefinitions>
+
+                                    <IconSourceElement Grid.Column="0"
+                                                       Width="16"
+                                                       Height="16"
+                                                       IconSource="{x:Bind Icon, Mode=OneWay, Converter={StaticResource IconSourceConverter}}" />
+
+                                    <TextBlock Grid.Column="1"
+                                               Text="{x:Bind Name}" />
+
+                                </Grid>
+                            </DataTemplate>
+                        </muxc:RadioButtons.ItemTemplate>
+                    </muxc:RadioButtons>
+                </local:SettingContainer>
+                <Button x:Name="DuplicateButton"

```

Comment:
If a control isn't used in the "code behind", you don't need to name it :smile:

Reply:
Actually, you _need_ this name. Please add the `AutomationProperty::Name` to it. You'll have to do this via the code. See https://github.com/microsoft/terminal/blob/361877cf1b1be2824f9f0373f1cf524b81cebb12/src/cascadia/TerminalSettingsEditor/Profiles.cpp#L419

Reply:
Added the automation property!